python/pyarrow/_dataset.pyx

```diff
@@ -1171,6 +1178,10 @@ cdef class CsvFileFormat(FileFormat):
     """
     cdef:
         CCsvFileFormat* csv_format
+        # The encoding field in ReadOptions does not exist in the C++ struct.
+        # We need to store it here and override it when reading
+        # default_fragment_scan_options.read_options
+        public ReadOptions read_options_py
```

Comment:
This does not need to be visible to the user, how about renaming it to stress it's an internal detail?
```suggestion
        public ReadOptions _read_options_py
```

Reply:
Thanks! changed in [b9982c8](https://github.com/apache/arrow/pull/13820/commits/b9982c8d28d898bbf506135bf53bad629728afef)